lib/maths/time_series/CTimeSeriesTestForSeasonality.cc

```diff
@@ -202,6 +203,33 @@ std::string CNewSeasonalComponentSummary::print() const {
     return result.str();
 }
 
+CSeasonalComponent
+CNewSeasonalComponentSummary::createSeasonalComponent(double decayRate,
+                                                      double bucketLength) const {
+    auto time = this->seasonalTime();
+    core_t::TTime period{time->period()};
+    core_t::TTime startTime{this->initialValuesStartTime()};
+    core_t::TTime endTime{this->initialValuesEndTime()};
+    core_t::TTime maxTimeShiftPerPeriod{
+        this->isOneOf(CNewSeasonalComponentSummary::E_Day | CNewSeasonalComponentSummary::E_Week |
+                      CNewSeasonalComponentSummary::E_Year)
+            ? 0
+            : this->bucketLength() / 2};
+    const auto& initialValues = this->initialValues();
+
+    // If we see multiple repeats of the component in the window we use
+    // a periodic boundary condition, which ensures that the prediction
+    // at the repeat is continuous.
+    auto boundaryCondition = period > time->windowLength()
+                                 ? common::CSplineTypes::E_Natural
+                                 : common::CSplineTypes::E_Periodic;
+    CSeasonalComponent component{
```

Comment:
Nit: hack so formatting doesn't get messed up:
```suggestion
    CSeasonalComponent component(
```

Reply:
changed